src/ObjWriting/XModel/Export/XModelBinWriter.cpp

```diff
@@ -0,0 +1,484 @@
+#include "XModelBinWriter.h"
+
+#pragma warning(push, 0)
+#include <Eigen>
+#pragma warning(pop)
+
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <lib/lz4.h>
+#include <ostream>
```

Comment:
You are using `std::ostringstream` which is defined in header `sstream`.
It is possible to sometimes use the standard lib features without including the appropriate headers because other headers that you included do include them.

These inclusions of your included headers may change though through changing unrelated header files, the standard lib version being used (version updates; MSVC STL vs glibc).

So to be safe across platforms and versions it is a good idea to always include all headers of all things you are making use of.

So i'd add the following headers:
```cpp
#include <format>
#include <limits>
#include <sstream>
#include <string>
```
These are the ones that I found being used in the file.